lib/ProductOpener/Ingredients.pm

```diff
@@ -2726,11 +2718,34 @@ sub get_missing_ciqual_codes ($ingredients_ref) {
 	return @ingredients_without_ciqual_codes;
 }
 
+=head2 estimate_ingredients_percent_service ( $product_ref )
+
+Compute minimum and maximum percent ranges and percent estimates for each ingredient and sub ingredient
+
+=cut 
+
+sub estimate_ingredients_percent_service ($product_ref) {
+
+	if (compute_ingredients_percent_min_max_values(100, 100, $product_ref->{ingredients}) < 0) {
+
+		# The computation yielded seemingly impossible values, delete the values
+		delete_ingredients_percent_values($product_ref->{ingredients});
+		$product_ref->{ingredients_percent_analysis} = -1;
+	}
+	else {
+		$product_ref->{ingredients_percent_analysis} = 1;
+	}
+
+	compute_ingredients_percent_estimates(100, $product_ref->{ingredients});
+
+	return;
```

Comment:
I think it would be good to return the specific properties that were modified, rather than modifying them in place, as that will make it easier to separate functions in the future